src/iop/tonecurve.c

```diff
@@ -344,6 +347,19 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
                                          : d->table[ch_b][CLAMP((int)(b_in * 0x10000ul), 0, 0xffff)]);
         }
       }
+      else if(autoscale_ab == DT_S_SCALE_MANUAL_LCH)
+      {
+        const float pi = 3.14159f;
+        const float C_in = sqrtf(in[1]*in[1] + in[2]*in[2]) / 181.019f;
+        const float h_in = ((in[1]!=0.0f ? atanf(in[2]/in[1]) : (in[2]>0.0f ? 0.5f : -0.5f) * pi) + ((in[1] < 0.0f) ? pi : ((in[2] < 0.0f) ? 2.0f * pi : 0.0f))) / (2.0f * pi);
+        float C_out = L_in == 0.0f ? C_in : C_in * d->table[ch_CL][CLAMP((int)(L_in * 0x10000ul), 0, 0xffff)] * 0.02f;
+        C_out = h_in == 0.0f ? C_out : C_out * d->table[ch_Ch][CLAMP((int)(h_in * 0x10000ul), 0, 0xffff)] * 0.02f;
```

Comment:
so you are shifting input hue by the same amount you shift the chroma ? Don't you want to preserve the hue untouched ? I don't understand what's going on here.

Reply:
Hue doesn't change. C is modified for the selected hue.